Sources/TuistCache/Cache/URL+Cache.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+import TuistCore
+
+enum CacheURLError: LocalizedError {
+    case missingCloudConfig
+    case incorrectCloudConfig
+}
+
+extension URL {
+    static func apiCacheURL(hash: String, config: Config, contentMD5: String? = nil) throws -> URL {
+        guard let cloudConfig = config.cloud else { throw CacheURLError.missingCloudConfig }
+        guard var urlComponents = URLComponents(url: cloudConfig.url, resolvingAgainstBaseURL: false) else {
+            throw CacheURLError.incorrectCloudConfig
+        }
+
+        urlComponents.path = "/api/cache"
+        var queryItems = [
+            URLQueryItem(name: "project_id", value: cloudConfig.projectId),
+            URLQueryItem(name: "hash", value: hash),
+        ]
+
+        if let contentMD5 = contentMD5 {
+            queryItems.append(URLQueryItem(name: "content_md5", value: contentMD5))
```

Comment:
I'd call this parameter "hash" to keep it agnostic of the implementation of the algorithm. Assuming we use only one now.

Reply:
There is already another parameter called hash, which, AFAIU, represents the "version" of the artefact to check/get/store.

This content_md5 is (will be) used to check the validity of the uploaded artefact. Google Cloud apparently forces us to give the md5 checksum of the file to upload, so they can validate it on their end.

That said, I'm happy to rename it with something you prefer, just with something different than `hash` as it's already used

Reply:
copying @pepibumur here, just to make sure that what I wrote is correct ;)

Reply:
Right. Hash and content MD5 are 2 different things:
- *Hash*: It's used to uniquely identify a resource in the cache.
- **Content MD5:** It's used to verify that the content uploaded to the storage is the one that was originally intended to be uploaded. This prevents man in the middle atacks.

Reply:
ok got it, it's the classic md5 checksum of the file. I thought we're taking about the hash :)